File: CS-Examples/03_Cells/CutCellsToOtherPosition.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

using Spire.Xls;

namespace CutCellsToOtherPosition
{
    public partial class Form1 : Form
    {

        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, System.EventArgs e)
        {
            //Create a workbook.
            Workbook workbook = new Workbook();

            //Load the document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\SampleB_2.xlsx");

            //Get the first worksheet
            Worksheet sheet = workbook.Worksheets[0];

            // Define the original source range to be copied (cells A1 to C5)
            CellRange Ori = sheet.Range["A1:C5"];

            // Define the destination range where the source range will be copied to (cells A26 to C30)
            CellRange Dest = sheet.Range["A26:C30"];

            //Copy the range to other position
            sheet.Copy(Ori, Dest, true, true, true);

            //Remove all content in original cells
            foreach (CellRange cr in Ori)
            {
                cr.ClearAll();
            }

            ///Specify the filename for the resulting Excel file
            string result = "result.xlsx";

            // Save the workbook to the specified file in Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // View file
            ExcelDocViewer(result);
        }
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }


}

File: CS-Examples/03_Cells/CutCellsToOtherPosition.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel

Imports Spire.Xls

Namespace CutCellsToOtherPosition
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            'Instantiate a new workbook object.
            Dim workbook As New Workbook()

            'Load an existing Excel document from the specified file path.
            workbook.LoadFromFile("..\..\..\..\..\..\Data\SampleB_2.xlsx")
            'Retrieve the first worksheet from the workbook.
            Dim sheet As Worksheet = workbook.Worksheets(0)
            'Define a cell range for the original cells.
            Dim Ori As CellRange = sheet.Range("A1:C5")
            'Define a cell range for the destination cells.
            Dim Dest As CellRange = sheet.Range("A26:C30")

            'Copy the content and formatting of the original cells to the destination cells, including values, formulas, and styles.
            sheet.Copy(Ori, Dest, True, True, True)

            'Iterate through each cell range within the original range.
            For Each cr As CellRange In Ori
                'Clear all content, including values, formulas, and formatting, in the current cell range.
                cr.ClearAll()
            Next cr

            'Specify the file name for the output file.
            Dim result As String = "result.xlsx"
            'Save the modified workbook to the specified output file using Excel 2010 version.
            workbook.SaveToFile(result, ExcelVersion.Version2010)
            ' Release the resources used by the workbook
            workbook.Dispose()

            ExcelDocViewer(result)
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class


End Namespace
